tools/harness-thci/OpenThread.py

```diff
@@ -245,27 +259,41 @@ def __getIp6Address(self, addressType):
         meshEIDAddr = ''
 
         addrs = self.__sendCommand('ipaddr')
+
+        # find rloc address firstly as a reference for current mesh local prefix as for some TCs,
```

Comment:
Thanks for fixing this address selection method, I can confirm it didn't work well in the master branch, due to case-sensitivity nature of `startswith`.

BTW. I have introduced a new CLI calls such as, `ipaddr rloc`, `ipaddr mleid` and `ipaddr linklocal`, so potentially you can use it to simplify the code even more.

Reply:
@LuDuda, Thanks for your confirmation. The second commit of this PR in which all addresses are transformed to be in lower cases is to resolve the case issue introduced in #3639.

It is a good idea to update the THCI to use the new CLIs. Will update soon.